packages/modal-ui/src/utils/UiUtil.ts

```diff
@@ -76,22 +76,20 @@ export const UiUtil = {
     }
   },
 
-  handleMobileLinking(wallet: WalletData) {
+  handleMobileLinking(wallet: WalletData, target = '_self' as '_self' | '_blank') {
     const { walletConnectUri } = OptionsCtrl.state
     const { mobile, name } = wallet
     const nativeUrl = mobile?.native
     const universalUrl = mobile?.universal
-
     UiUtil.setRecentWallet(wallet)
-
     function onRedirect(uri: string) {
-      let href = ''
       if (nativeUrl) {
-        href = CoreUtil.formatUniversalUrl(nativeUrl, uri, name)
+        const href = CoreUtil.formatNativeUrl(nativeUrl, uri, name)
+        CoreUtil.openHref(href, target)
```

Comment:
Seems like the `openHref` params are the same for each case, why not call it outside of the if/else clauses?

Reply:
they are but it adds additional condition as we need to check that href exists. lets leave it as it